drivers/serial/uart_sifive.c

```diff
@@ -8,6 +8,8 @@
  * @brief UART driver for the SiFive Freedom Processor
  */
 
+#define DT_DRV_COMPAT sifive_uart0
```

Comment:
Also, what if the driver supports two compatibles? 
Are there some usage rules?

Reply:
> Also, what if the driver supports two compatibles?
> Are there some usage rules?

There are some limitations in this regard. https://github.com/zephyrproject-rtos/zephyr/issues/19904

Reply:
That being said, I think at an API level there's nothing stopping a driver from making their own instance wrappers for each compatible, once the underlying issues with the bindings are resolved. The `DT_N_INST_i_compat` macro namespaces will not need to change.

Reply:
I have several cases in mind, need deeper study to see impacts